packages/builder/src/definition.ts

```diff
@@ -104,7 +104,7 @@ export class ChainDefinition {
 
           for (const output of actionOutputs) {
             debug(`deps: ${fullActionName} provides ${output}`);
-            if (!this.dependencyFor.has(output)) {
+            if (!this.dependencyFor.has(output) || action === 'setting' || action === 'var') {
```

Comment:
This wouldn't work weird when having a `var` that uses another `var`? Meaning that it wont be able to find correctly the dependency?